File: java/src/edu/caltech/ipac/firefly/visualize/draw/DrawSymbol.java
package edu.caltech.ipac.firefly.visualize.draw;
/**
 * User: roby
 * Date: 5/8/12
 * Time: 2:17 PM
 */


import edu.caltech.ipac.util.StringUtils;

/**
* @author Trey Roby
*/
public enum DrawSymbol {
    X,SQUARE,CROSS,DIAMOND,DOT,CIRCLE, SQUARE_X, EMP_CROSS,EMP_SQUARE_X;


    public static DrawSymbol getSymbol(String s) {
        DrawSymbol retval= DrawSymbol.X;
        if (!StringUtils.isEmpty(s)) {
            s= s.toUpperCase();
            retval= Enum.valueOf(DrawSymbol.class, s);
        }
        return retval;
    }

}

File: src/firefly/js/visualize/draw/DrawSymbol.js
import Enum from 'enum';

/**
 * @typedef {Object} DrawSymbol
 * enum can be one of
 * @prop X
 * @prop SQUARE
 * @prop CROSS
 * @prop DIAMOND
 * @prop DOT
 * @prop CIRCLE
 * @prop SQUARE_X
 * @prop EMP_CROSS
 * @prop EMP_SQUARE_X
 * @prop BOXCIRCLE
 * @prop ARROW
 * @prop TEXT
 * @type {Enum}
 */

/** @type DrawSymbol */
export const DrawSymbol = new Enum([
    'X','SQUARE','CROSS','DIAMOND','DOT','CIRCLE', 'SQUARE_X', 'EMP_CROSS','EMP_SQUARE_X',
    'BOXCIRCLE', 'ARROW', 'ROTATE', 'POINT_MARKER', 'TEXT'
], { ignoreCase: true });

